CIP-XXXX/CIP-XXXX.md

```diff
@@ -0,0 +1,180 @@
+
+Retrieves the list of extensions enabled by the wallet. This may be influenced by the set of extensions requested in the initial `enable` request.
+
+## Rationale: how does this CIP achieve its goals?
+
+This API does not offer *useful* functionality for dApp-Wallet interactions; rather, all useful functionality should be offered through extensions. By keeping this API as a skeleton, we aim to avoid the chance of introducing functionality we come to regret later. Instead, this proposal should be the minimum foundation required to build on top of.  
+
+The alternative to this proposal is the addition of extensions to CIP-30's specification. This would mean that the CIP-30 API would be included in all extended dApp-Wallet connections. With the current landscape of dApps and wallets this seems fine, but it is uncertain if we want to indefinitely support the CIP-30 API.
+
+By stripping functionality from this API we facilitate dApp-Wallet connections with less functions than CIP-30 API. A use case for this is connections where minimal wallet information is wished to be shared with a dApp.
+
+Here we attempt to mitigate the impact of succeeding CIP-30 as *the* Cardano wallet connector, by enabling it's functionality by default at connection time. This should minimize the impact this proposal could have on established dApps who support CIP-30. By only adding what is necessary for extensions we aim to reduce the required load for wallets to adopt this standard.
+
+### Extensions
+
+Extensions provide an extensibility mechanism and a way to negotiate (possibly conflicting) functionality between a dApp and a wallet provider. There's rules enforced as for what extensions a wallet decide to support or enable. The current mechanism only gives a way for wallets to communicate their choice back to a dApp.
+
+We use object as extensions for now to leave room for adding fields in the future without breaking all existing interfaces. At this point in time however, objects are expected to be singleton.
+
+Extensions can be seen as a smart versioning scheme. Except that, instead of being a monotonically increasing sequence of numbers, they are multi-dimensional feature set that can be toggled on and off at will. This is a versioning "à-la-carte" which is useful in a context where:
+
+1. There are multiple concurrent standardization efforts on different fronts to accommodate a rapidly evolving ecosystem;
+2. Not everyone agrees and has desired to support every existing standard;
+3. There's a need from an API consumer standpoint to clearly identify what features are supported by providers.
+
+By moving CIP-30 style modifications into a formalized extension scheme we aim to avoid fragmentation of functionality between different dApp-Wallet bridges.
+
+## Path to Active
+
+### Acceptance Criteria
+
+- [ ] The interface is implemented and supported by various wallet providers.
+- [ ] The interface is used by dApps to interact with wallet providers.

```

Comment:
This is supposed to be backwards-compatible with cip30, so I believe this criteria is already satisfied.
```suggestion
- [x] The interface is implemented and supported by various wallet providers.
- [x] The interface is used by dApps to interact with wallet providers.

```

Reply:
Wallets will still have to implement the additional extension scheme functions. Although this change should be transparent to CIP-30 compatible dApps.

Reply:
Yeah, it's probably up to CIP editors to set this criteria